### PR TITLE
Add some class to work with vectors

### PR DESCRIPTION
**Pull request by nicoxxl**
- BaseVector is the base class meant to create other vector class
 - CartesianVector is a class for nth dimensional cartesian vectors
 - Vector3 is a class which look like Vec3, but since Vec3 is reused, I didn't take any risk

The goal is to simplify working with coordinates, for example :
```python
me = Vector3(0, 10, 0) # Where I am
target = Vector3(5, 12, 10) # the position of the target
to_target = target - me
to_target.norm() # Distance from the target
to_target.yaw_pitch() # The yaw&pitch needed to look at the target
```

### CODE REVIEW

**Comment by Gjum** [comment]
This has been for discussion for a while (#47), as the Vec3 class is little more than a wrapper around `x, y, z` for packet sending.
I really like the idea of yaw/pitch conversion.
My only concern is if all the high-level functionality is really needed, but other projects using Spock might profit from that.

**Comment by nickelpro** [comment]
I like it, but none of my code uses it. gjum review/merge is on you

File: spock/utils.py [hunk 1]
```diff
@@ -45,6 +46,188 @@ def add_vector(self, x=None, y=None, z=None, vec=None):
...
+	def __add__(self, other):
+		return self.__class__(*map(sum, zip(self, other)))
+
+	__iadd__ = __add__
+
+	def __neg__(self):
+		return self.__class__(*map(lambda a:-a, self))
+
+	def __sub__(self, other):
+		return self.__class__(*map(lambda a:a[0] - a[1], zip(self, -other)))
```
Comment (Gjum) [inline spock/utils.py]: Did you test this? It seems like some double-negation is going on here...

File: spock/utils.py [hunk 2]
```diff
@@ -45,6 +46,188 @@ def add_vector(self, x=None, y=None, z=None, vec=None):
...
+		return self.vector[item]
+
+	def __len__(self):
+		return len(self.vector)
+
+
+class CartesianVector(BaseVector):
+
+	# Math operations
+	# Is __abs__ really useful ?
```
Comment (Gjum) [inline spock/utils.py]: What about using abs for norm?

**Comment by Gjum** [comment]
I'll try to use this in `physics`. If it proves usable (and I think it will), I'll merge.

File: spock/utils.py (hunk 2 above)
Comment (nicoxxl) [inline spock/utils.py]: When I wrote this code, I make it super generic, but it is your project and I will trust you.

**Pull request merged**